website/static/js/filepage/index.js

```diff
@@ -53,7 +53,15 @@ var SharePopover =  {
                         m('li', m('a[href="#embed"][data-toggle="tab"]', 'Embed'))
                     ]), m('br'),
                     m('.tab-content', [
-                        m('.tab-pane.active#share', m('.input-group', [
+                        m('.tab-pane.active#share', {
+                            config: function(element) {
+                                if ($(element).find('.shareButtons').length === 0) {
+                                    var shareContents = $('#shareContentHolder').html();
```

Comment:
Manipulating the DOM manually like this should be unnecessary. Also, this approach has duplication--the HTML for the social media buttons need to be copied into every template they're used.

Since mithril is already used here, why not create a mithril component for the SHARE buttons?

```javascript
// in js/components/socialShare.js
var ShareButtons = {
    view: function(ctrl, options) {
        return m('div.share-buttons', {}, [
            m('a', {href: 'https://twitter...'})
            // ...
        ])
    }
}

// filepage/index.js
m('.tab-page.active#share', {}, [
    m.component(ShareButtons, {})
])
```